### PR TITLE
avoid use of undefined ABSL_HAVE_ELF_MEM_IMAGE

### PR DESCRIPTION
**Pull request by christoph-cullmann**
this avoids e.g. on macOS:

In file included from abseil-cpp/absl/debugging/internal/elf_mem_image.cc:18:
abseil-cpp/absl/debugging/internal/elf_mem_image.h:37:5: error: 'ABSL_HAVE_ELF_MEM_IMAGE' is not defined, evaluates to 0 [-Werror,-Wundef]
    ^
1 error generated.

### CODE REVIEW

**Comment by christoph-cullmann** [comment]
I signed it!

**Comment by vslashg** [comment]
It doesn't appear from our end that you signed the CLA.  Could you take another look at this?

File: absl/debugging/internal/elf_mem_image.h
Comment (zhangxy988) [inline absl/debugging/internal/elf_mem_image.h]: `#if defined(ABSL_HAVE_ELF_MEM_IMAGE)` is good enough: the macros should either be 1 or undefined.

**Review by zhangxy988 — CHANGES_REQUESTED** [review]

**Comment by christoph-cullmann** [comment]
I will adapt the request and I assume my e-mail address doesn't match the CLA, I will fix that, too.

**Comment by christoph-cullmann** [comment]
CLA should now be ok and matching my mail address.
Only checking for "is defined" now.

File: absl/debugging/internal/elf_mem_image.h
Comment (christoph-cullmann) [reply absl/debugging/internal/elf_mem_image.h]: Resolved by using #ifdef like above in the file.

**Review by zhangxy988 — APPROVED** [review]
Please wait for our release manager to merge and sync with our internal code base.

**Comment by christoph-cullmann** [comment]
Sure, thanks for taking care.
I assume I will have a few other warning fixes in the future as we use rather strict compiler flags for the code using the abseil headers (together with MSVC).

**Comment by zhangxy988** [comment]
Sorry for the delay. I am the release manager this week and I will test this change against our internal code base asap.

**Pull request merged**
